ehr/src/org/labkey/ehr/utils/TriggerScriptHelper.java

```diff
@@ -1215,30 +1237,106 @@ private String checkOtherDrawsQuantity(String id, Date date, String rowObjectId,
             }
+            else if (getThresholdCheck())
+            {
+                //only report about problematic bloods nearing the limit in the current transaction
+                //because allBloods contains everything and we want to distinguish them
+                if (blood1.getIsInTransaction())
+                {
+                    double maxAllowableThreshold = maxAllowable - getThresholdLimit();
+                    if (bloodNextInterval > maxAllowableThreshold)
+                    {
+                        closeToThreshold.add(bloodNextInterval);
+                    }
+                }
+            }
         }
 
-        //always report the most severe overage
-        if (!overages.isEmpty())
+        if (!overages.isEmpty() || !closeToThreshold.isEmpty())
+        {
+            StringBuilder errorMsgBuilder = new StringBuilder();
+
+            //always report the most severe overage
+            if (!overages.isEmpty())
+            {
+                errorMsgBuilder.append("Blood volume of ");
+                errorMsgBuilder.append(rowQuantity);
+                errorMsgBuilder.append(" (");
+                errorMsgBuilder.append(overages.descendingSet().iterator().next());
+                errorMsgBuilder.append(" over ");
+                errorMsgBuilder.append(interval);
+                errorMsgBuilder.append(" days) exceeds the allowable volume of ");
+                errorMsgBuilder.append(maxAllowable);
+                errorMsgBuilder.append(" mL (weight: ");
+                errorMsgBuilder.append(weight);
+                errorMsgBuilder.append(" kg).\n");
+            }
+
+            //report all the ones that are close to limit
+            for (double closeValue : closeToThreshold.descendingSet())
+            {
+                errorMsgBuilder.append("Limit notice! Blood volume of ");
+                errorMsgBuilder.append(rowQuantity);
+                errorMsgBuilder.append(" (");
+                errorMsgBuilder.append(closeValue);
+                errorMsgBuilder.append(" over ");
+                errorMsgBuilder.append(interval);
+                errorMsgBuilder.append(" days) is within ");
+                errorMsgBuilder.append(getThresholdLimit());
+                errorMsgBuilder.append(" mL of the max allowable limit of ");
+                errorMsgBuilder.append(maxAllowable);
+                errorMsgBuilder.append(" mL (weight: ");
+                errorMsgBuilder.append(weight);
+                errorMsgBuilder.append(" kg).\n");
+            }
+
+            return errorMsgBuilder.toString();
+        }
+        else
         {
-            return "Blood volume of " + rowQuantity + " (" + overages.descendingSet().iterator().next() + " over " + interval + " days) exceeds the allowable volume of " + maxAllowable + " mL (weight: " + weight + " kg)";
+            return null; // No errors
         }
+    }
+
+    public double getThresholdLimit()
```

Comment:
I am guessing maybe this is how you intend to let WNPRC-specific code set this value? If so, I suggest making it more descriptive and adding javadoc to make the purpose of this method clearer. Same for any other public methods you add to this class

Reply:
I can add a javadoc

Reply:
Thanks. Documention is probably most important for getThresholdCheck(), and I would really rename the latter to something much more explanatory. TriggerHelper isnt blood-specific, and getThresholdCheck doesnt say anything about being blood-related. I would even consider doWarnForBloodNearOverages() or something very clear.

Reply:
Added a javadoc and a new name for getting the limit: getBloodNearingOveragesThreshold() based on setting custom props, what do you think about using _centerCustomProps?